### ISSUE TITLE
Help - 'Load more' on iframes

### ISSUE THREAD

**Issue by steviecodemuncher** [webcreate/infinite-ajax-scroll#307]
Hi,

Great lazy load solution by the way, thank you!

I have a situation whereby the iframe does not know when a user has scrolled to the bottom, therefore the 'load more items' link does not appear.

As a way around this, is there a way by editing the jquery file so that the link is always there rather than fading in and appearing as it does normally when the bottom is reached?

Unless you can suggest a better way for iframe / infinate-ajax-scroll code to work together - I think this is a good work-around?

I'd really appreciate your help with this! 

Many thanks!

**Comment by fieg** [comment]
Hi 👋, thanks for reporting your issue. 

As far as I know Infinite Ajax Scroll should work in an iframe. Do you have a live url so I can have a look?

**Comment by steviecodemuncher** [comment]
Hi, thanks for the quick response!

Here's a link with the iframe and Ajax scroll code in place (not working):
https://icomparebroadband.co.uk/broadband/deals/

And here's a direct link to the page of the iframe (Ajax scroll working. However, the link needs to be clicked twice for some reason?):
https://icomparebroadband.co.uk/deals/search.php?&category=&q=Broadband+only&order=asc

We use a plugin that auto-adjusts the height as it's dynamically created results. I suspect that maybe causing the issue? That's why if there's no other solution, the suggestion I made would be OK for us!

Cheers!

**Comment by fieg** [comment]
My internet is pretty crappy here at the moment, so I have a look later on, but the first thing I noticed is that the IASTriggerExtension is called twice. That's probably why you have to click twice.

```js
ias.extension(new IASTriggerExtension({offset: 1}));
ias.extension(new IASTriggerExtension({text: 'Load more items'}));
```

Should become:

```js
ias.extension(new IASTriggerExtension({offset: 1, text: 'Load more items'}));
```

**Comment by steviecodemuncher** [comment]
Oh, right ha ha, whoops!

OK no worries, thanks again

**Comment by fieg** [comment]
I looked into it some more. The auto-adjusting plugin is indeed causing IAS not to detect any scrolling. 

Perhaps you can try to add a link/button below the items container and call [next](https://infiniteajaxscroll.com/docs/methods.html#next) on click. You can also listen to the [`noneLeft`](https://infiniteajaxscroll.com/docs/events.html#noneleft) event and remove the link/button.

To give you an idea:

```html
<div id="container">
   <!-- here are the items appended -->
</div>
<div>
  <!-- this gets pushed down as new items are appended -->
  <a onclick="$.ias('next')" id="my_next_link">Show more</a>
</div>
```

And remove the link when there are no more pages left:

```js
ias.on('noneLeft', function() { $('#my_next_link').remove(); });
```

Hope this helps!

**Comment by steviecodemuncher** [comment]
Thanks!

I got the 'Show more' working, but removing the last one is an issue. It only disappears when clicked?

I tried removing the other nonLeft condition :    ias.extension(new IASNoneLeftExtension({text: ""})); but it still gets called??

Cheers, massive help!

**Comment by fieg** [comment]
Hm, it looks like the noneLeft event comes one page too late. Looking at the code this also seems illogical, so I'm working on a fix: #308. 

I'm planning on releasing a new version later today, so it should include this fix as well.

**Comment by steviecodemuncher** [comment]
Great, thanks! Keep us posted 👍

**Comment by fieg** [comment]
New version is out. You can download it [here](https://infiniteajaxscroll.com/jquery-ias.min.js). Let me know if this helps.

**Comment by steviecodemuncher** [comment]
Hi!

Thanks for the update!

I tested the new version and it does remove the last instance of the link. However, when I tested the webpage on Apple iPhone/iPad's I noticed that the number/row count for results returned jumped to include the next page (without clicking any link) i.e. 20 instead of 10.

I then returned to the older version, cleared all caches, but it seams that this issue is an old one.

Any thoughts??
